### PR TITLE
feat(swarm)!: remove unused `PendingConnectionError::Io` variant

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

This variant is never constructed.

## Notes

Discovered whilst trying to sort out the error situation in #3254.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,10 @@
   If you have previously set `connection_event_buffer_size` you should re-evaluate what a good size for a _per connection_ buffer is.
   See [PR 3188].
 
+- Remove `PendingConnectionError:::IO` variant.
+  This was never constructed.
+  See [PR XXXX].
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
  See [PR 3373].
```

File: swarm/CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +35,7 @@
 [PR 3272]: https://github.com/libp2p/rust-libp2p/pull/3272
 [PR 3327]: https://github.com/libp2p/rust-libp2p/pull/3327
 [PR 3188]: https://github.com/libp2p/rust-libp2p/pull/3188
+[PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
[PR 3373]: https://github.com/libp2p/rust-libp2p/pull/3373
```

**Review by mxinden — APPROVED** [review]
Good catch.

**Comment by mxinden** [comment]
Adding send-it label. Given that this is not a draft pull request and given that it targets the `master` branch, I am assuming that this is not a "stacked pull request".

**Comment by thomaseizinger** [comment]
> Adding send-it label. Given that this is not a draft pull request and given that it targets the `master` branch, I am assuming that this is not a "stacked pull request".

Correct, there are more on top of this one :)

**Comment by thomaseizinger** [comment]
@mergifyio refresh

**Comment by thomaseizinger** [comment]
@mergifyio refresh

**Comment by thomaseizinger** [comment]
@mergifyio refresh

**Pull request merged**